### ISSUE TITLE
how to rename the uploaded file and sent as response ?

### ISSUE THREAD

**Issue by umesh0492** [strongloop/loopback-component-storage#36]

**Comment by moguelator** [comment]
I think it is not posible to do this at the moment. Have a look at issue #20.

**Comment by shawnrice** [comment]
For now, you can try to hack the component a bit.

The way that I'm currently doing it is to alter 
1. `lib/providers/filesystem/index.js`, specifically, the `FileSystemProvider.prototype.upload ` function around line 160. 
2. You'll also have to alter `lib/storage-handler.js` around line 60. The current line in there is
````
var writer = provider.upload({container: container, remote: part.filename});
````
Use that to pass whatever arguments you want into the previous function.
3. You'll have to find some way to pass the arguments to the function. You can try burying them in the request object because everything else that you might send seems to get stripped out. I haven't found a satisfactory way alter the model.

But do keep in mind that if you do the above, then it _might_ break the functionality for the cloud services. Also, it's probably not the most secure way of managing it, so make sure you aren't uploading anything sensitive.

**Comment by seriousben** [comment]
see #20

**Comment by seriousben** [comment]
Fixed by #41
